### PR TITLE
[#319] Remove `atoi` from `pgagroal-cli`

### LINKED ISSUE #319: Remove `atoi` from `pgagroal-cli`
In case of remote connection, `pgagroal-cli` uses `atoi` to get the port number <https://github.com/agroal/pgagroal/blob/master/src/cli.c#L451>.
This should be removed in favor of `strtol` or similar.

### PR DESCRIPTION
**Pull request by fluca1978**
When doing a remote connection to the pooler, the user has to specify
the port with `-p` flag. The value is converted with `strtol` and the
ranges are checked.
Also, since pgagroal cannot be run as root, it cannot bind to ports
lower than 1024, therefore if the user specifies a wrong port number,
it gets a different error message rather than "No route to host".

Close #319

### CODE REVIEW

File: src/cli.c [hunk 1]
```diff
@@ -448,9 +450,27 @@ main(int argc, char** argv)
          else
          {
             /* Remote connection */
-            if (pgagroal_connect(host, atoi(port), &socket))
+
+            l_port = strtol(port, NULL, 10);
+            if ((errno == ERANGE
+                 && (l_port == LONG_MAX || l_port == LONG_MIN))
+                || (errno != 0 && l_port == 0))
```
Comment (jesperpedersen) [inline src/cli.c:457]: Could you put `&&` and `||` on the previous lines ?

File: src/cli.c [hunk 2]
```diff
@@ -448,9 +450,27 @@ main(int argc, char** argv)
...
+            {
+               warnx("Specified port %s out of range", port);
+               goto done;
+            }
+
+            // cannot connect to port less than 1024 becayse pgagroal
```
Comment (jesperpedersen) [inline src/cli.c:463]: Spelling

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/cli.c (hunk 1 above)
Comment (fluca1978) [reply src/cli.c]: fixed

File: src/cli.c (hunk 2 above)
Comment (fluca1978) [reply src/cli.c]: fixed

**Comment by jesperpedersen** [comment]
Merged.

Thanks for the contribution !

**Pull request closed without merging**